packages/aragon-client/src/index.js

```diff
@@ -36,6 +36,45 @@ class AppProxy {
     ).pluck('result')
   }
 
+  /**
+   * Create a handle to an external contract.
+   *
+   * @param  {string} address The address of the external contract
+   * @param  {Array<Object>} jsonInterface The JSON interface of the external contract
+   * @return {Object}
+   */
+  external (address, jsonInterface) {
+    let contract = {
+      events: (fromBlock = 0) => {
+        return this.rpc.sendAndObserveResponses(
+          'external_events',
+          [
+            address,
+            jsonInterface.filter(
+              (item) => item.type === 'event'
+            ),
+            fromBlock
+          ]
+        ).pluck('result')
+      }
+    }
+
+    // Bind calls
+    const callMethods = jsonInterface.filter(
+      (item) => item.type === 'function' && item.constant
+    )
+    for (let methodABI of callMethods) {
```

Comment:
I do find it slightly odd this isn't a `forEach` or `reduce`

Reply:
could be a `forEach`, probably also a reduce but would feel a bit weird with reduce imo